snapcraft.yaml

```diff
@@ -1120,7 +1120,7 @@ parts:
   zfs-2-2:
     source: https://github.com/openzfs/zfs
     source-depth: 1
-    source-tag: zfs-2.2.3
+    source-tag: zfs-2.2.4
```

Comment:
We discussed in the stand up that it would be good to perform more verification on these versions, e.g.:
- Are there any outstanding security issues?
- Can we make snapcraft verify a hash/GPG after downloading it?

Reply:
It seems that snapcraft supports `source-checksum` which accepts `algorithm/digest`.

https://snapcraft.io/docs/snapcraft-parts-metadata#heading--source

I'm not sure if this works with git repo, but maybe we can use the checksum of the source code archive?

Reply:
Thanks for bringing that up it makes sense. This `source-checksum` seems like the way to go, could be saved for another PR too.

Reply:
yes this isnt relevant for this PR - also a question I have is which hash is used for the source hash?

Reply:
Good point re the checksum being for the tarball. When we take care of this in another PR, `source-commit` could be used to reference the full SHA1 hash of the commit associated with the tag. In fact we might want to remove `source-tag` altogether and use:

```yaml
  zfs-2-2:
    source: https://github.com/openzfs/zfs
    source-depth: 1
    source-commit:  2566592045780e7be7afc899c2496b1ae3af4f4d # zfs-2.2.4
```

Reply:
That seems confusing to me - why not let the publisher decide which commit matches to a version?

Reply:
While I agree that's less readable, it protects us from any modifications of the tag. GitHub recommends doing so for 3rd party actions for example: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

https://michaelheap.com/ensure-github-actions-pinned-sha/ goes into more details about this ^.

Reply:
OK makes sense. Thanks for the links! but yes lets do this for all deps in a separate PR and not conflate them.

Reply:
Good. I'll also put up a Jira card to do the same with our various GH Actions.

Reply:
> Good. I'll also put up a Jira card to do the same with our various GH Actions.

May be worth investigating Renovate options for GH actions (which we use in quite few repositories). Renovate allows pinning the GH action version - will update commit hash of the GH action and add a comment next to it containing the actual version the hash refers to.

It is one of these two options:
- https://docs.renovatebot.com/presets-helpers/#helperspingithubactiondigeststosemver
- https://docs.renovatebot.com/presets-helpers/#helperspingithubactiondigests